index/scorch/segment/zap/count.go

```diff
@@ -53,3 +53,14 @@ func (c *CountHashWriter) Count() int {
 func (c *CountHashWriter) Sum32() uint32 {
 	return c.h.Sum32()
 }
+
+// Crc32Writer wraps an io.Writer by updating an IEEE crc32
+type Crc32Writer struct {
+	w   io.Writer
+	crc uint32
+}
+
+func (c *Crc32Writer) Write(b []byte) (int, error) {
+	c.crc = crc32.Update(c.crc, crc32.IEEETable, b)
+	return c.w.Write(b)
+}
```

Comment:
it's not clear to me why we need this instead of the CountHashWriter

Reply:
Biggest issue was a limitation of golang's hash/crc32 API, where crc32.Update() is a slower pathway that made me loathe to want to slow down CountHashWriter.

Reply:
Also, usage-wise, CountHashWriter starts a clean hash, where Crc32Writer is able to take a starting crc32 value.

Reply:
Actually, looking more closely at the crc32 codebase, looks like crc32.Update() isn't necessarily slower than a Write().  I'll take a look instead into modifying CountHashWriter to use crc32.Update() instead of Write().